perfaware/sim86/shared/Sim8086.cpp

```diff
@@ -0,0 +1,162 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <string>
+#include <sstream>
+#include "sim86_shared.h"
+#pragma comment (lib, "sim86_shared_debug.lib")
+#define ArrayCount(Array) (sizeof(Array) / sizeof((Array)[0]))
+
+s16 Registers[15] = {};
+
+char const* RegisterNames[][3] =
+{
+    {"", "", ""},
+    {"al", "ah", "ax"},
+    {"bl", "bh", "bx"},
+    {"cl", "ch", "cx"},
+    {"dl", "dh", "dx"},
+    {"sp", "sp", "sp"},
+    {"bp", "bp", "bp"},
+    {"si", "si", "si"},
+    {"di", "di", "di"},
+    {"es", "es", "es"},
+    {"cs", "cs", "cs"},
+    {"ss", "ss", "ss"},
+    {"ds", "ds", "ds"},
+    {"ip", "ip", "ip"},
+    {"flags", "flags", "flags"}
+};
+
+void SimulateInstruction(instruction Instruction)
+{
+    const char* Op = Sim86_MnemonicFromOperationType(Instruction.Op);
+    
+    if (Instruction.Op == Op_mov)
+    {
+        instruction_operand Dest = Instruction.Operands[0];
+        instruction_operand Source = Instruction.Operands[1];
+        s32 Value; // s32 to avoid issues on the bitshift right
+        switch (Source.Type)
+        {
+            case Operand_Register:
+            {
+                Value = Registers[Source.Register.Index];
+                if (Source.Register.Offset == 1) // Accessing high 8 bits
+                {
+                    Value >>= 8;
+                }
+                else if (Source.Register.Count == 1) // Accessing low 8 bits
+                {
+                    Value &= 0xFF;
+                }
+            } break;
+            case Operand_Immediate:
+            {
+                Value = Source.Immediate.Value;
+            } break;
+        }
+
+        Value = (s16)Value;
+
+        switch (Dest.Type)
+        {
+            case Operand_Register:
+            {
+                if (Dest.Register.Offset == 1) // Writing high 8 bits
+                {
+                    s16 Temp = Registers[Dest.Register.Index];
+                    Temp &= 0xFF;
+                    Value <<= 8;
+                    Registers[Dest.Register.Index] = Temp | Value;
+                }
+                else if (Dest.Register.Count == 1) // Accessing low 8 bits
+                {
+                    Value &= 0xFF;
+                    s16 Temp = Registers[Dest.Register.Index];
+                    Temp &= 0xFF00;
+                    Registers[Dest.Register.Index] = Temp | Value;
+                }
+                else
+                {
+                    Registers[Dest.Register.Index] = Value;
+                }
+            } break;
+        }
+
+    }
+
```

Comment:
There are a couple of empty lines here. I don't think they're useful, but maybe it's on purpose?

Reply:
Fixed 88f30e2f6f35580c958f66952b89903db207dbda